### PR TITLE
chore(eks): support Inf1 instances

### PR DESCRIPTION
**Pull request by pahud**
chore(eks): support inf1 instances

Amazon EKS supports Inferentia Instances today(see [Infrerentia Support](https://docs.aws.amazon.com/eks/latest/userguide/inferentia-support.html) and the doc [update](https://twitter.com/aws_doc/status/1268567994394968066)). This PR allows to create Amazon EKS worker nodes with `Inf1` instances. 

- Add `Inf1` instances support

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by pahud** [comment]
```ts
    cluster.addCapacity('Nodes', {
      instanceType: new ec2.InstanceType('inf1.2xlarge'),
      minCapacity: 1,
      spotPrice: '0.60',
    });
```
圖片

圖片


Love it!

**Comment by pahud** [comment]
Hi @eladb 

According to the [document](https://docs.aws.amazon.com/eks/latest/userguide/inferentia-support.html), when launching the `Inf1` instances into the nodegroup,  the `eksctl` will auto install the `neuron-device-plugin` daemon set
https://github.com/weaveworks/eksctl/blob/master/pkg/addons/assets/neuron-device-plugin.yaml

I believe we could build similar experience like this:

```ts
   // create a self-managed nodegroup with inf1.2xlarge instance(s)
    cluster.addCapacity('InfNodes', {
      instanceType: new ec2.InstanceType('inf1.2xlarge'),
      minCapacity:  1,
      spotPrice: '0.60',
      bootstrapOptions: {
        kubeletExtraArgs: '--node-labels neuron-device-plugin=enabled'
      }
    });

    // install the neuron-device-plugin daemonset
    cluster.addResource('NeuronDevicePlugin', addon.NeuronDevicePlugin.load());
```
The difference is that user can decide which `addon(s)` to be loaded and potentially have the freedom to customize the addon before loading it.

Behind the scene, all `inf1.2xlarge` instances will have `neuron-device-plugin=enabled` nodel label and the vended addon will have nodeSelector like this. So we can ensure the neuron device plugin ds will only be deployed on inf1 instances.

```ts
      spec: {
        nodeSelector: {
          "neuron-device-plugin": "enabled"
        },
```

What do you think?

**Comment by eladb** [comment]
@pahud what other Addons do you envision? Why not just hard-code support for the neuron device plugin when users add an inf1 instance?

File: packages/@aws-cdk/aws-eks/lib/cluster.ts [hunk 1]
```diff
@@ -1051,6 +1051,7 @@ export class EksOptimizedImage implements ec2.IMachineImage {
     this.amiParameterName = `/aws/service/eks/optimized-ami/${this.kubernetesVersion}/`
       + ( this.nodeType === NodeType.STANDARD ? 'amazon-linux-2/' : '' )
       + ( this.nodeType === NodeType.GPU ? 'amazon-linux-2-gpu/' : '' )
+      + (this.nodeType === NodeType.INFERENTIA ? 'amazon-linux-2-gpu/' : '')
```
Comment (eladb) [inline packages/@aws-cdk/aws-eks/lib/cluster.ts:1054]: Why do we need a separate node type here? Can't we just identify `inf1` instances as a gpu?

Comment (pahud) [reply packages/@aws-cdk/aws-eks/lib/cluster.ts:1054]: We just announced the **Amazon ECS-optimized Amazon Linux 2 (Inferentia) AMI**, which has different parameter store path and AMI value from the GPU one.  I guess we probably will have similar AMI branch for Amazon EKS

https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-optimized_AMI.html

Comment (pahud) [reply packages/@aws-cdk/aws-eks/lib/cluster.ts:1054]: There're some device plugins such as

https://github.com/NVIDIA/k8s-device-plugin
and
https://github.com/awslabs/aws-virtual-gpu-device-plugin

And I agree we hard-code it for this support to make it simple.

File: packages/@aws-cdk/aws-eks/lib/cluster.ts [hunk 2]
```diff
@@ -537,6 +542,10 @@ export class Cluster extends Resource implements ICluster {
       machineImageType: options.machineImageType,
     });
 
+    if (!this._neuronDevicePlugin && nodeTypeForInstanceType(options.instanceType) === NodeType.INFERENTIA) {
```
Comment (eladb) [inline packages/@aws-cdk/aws-eks/lib/cluster.ts:545]: Am I missing something? This is the constructor. By definition `_neuronDevicePlugin` would be `undefined` at this point, no?

```suggestion
    if (nodeTypeForInstanceType(options.instanceType) === NodeType.INFERENTIA) {
```

File: packages/@aws-cdk/aws-eks/lib/addons/assets/neuron-device-plugin.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
+# source: https://github.com/aws/aws-neuron-sdk/blob/master/docs/neuron-container-tools/k8s-neuron-device-plugin.yml
```
Comment (eladb) [inline packages/@aws-cdk/aws-eks/lib/addons/assets/neuron-device-plugin.yaml:1]: move from `addons/assets` to `addons` the deep nesting is not needed.

**Pull request merged**